packages/react-dom/src/ReactDOMSharedInternals.js

```diff
@@ -7,29 +7,18 @@
  * @flow
  */
 
-import {batchedUpdates} from 'react-reconciler/src/ReactFiberReconciler';
-import {
-  enqueueStateRestore,
-  restoreStateIfNeeded,
-} from './events/ReactDOMControlledComponent';
-import {
-  getInstanceFromNode,
-  getNodeFromInstance,
-  getFiberCurrentPropsFromNode,
-} from './client/ReactDOMComponentTree';
+import Dispatcher from './shared/ReactDOMDispatcher';
 
-const Internals = {
+type InternalsType = {
+  usingClientEntryPoint: boolean,
+  Events: mixed,
+  Dispatcher: typeof Dispatcher,
+};
+
+const Internals: InternalsType = {
   usingClientEntryPoint: false,
-  // Keep in sync with ReactTestUtils.js.
-  // This is an array for better minification.
-  Events: [
```

Comment:
Yeah I was trying to work around an issue where floatclient gets imported before this resolves so it can’t set the default dispatcher. I could solve that other ways but was mostly just trying get the shared internals to have complex imports